src/client/batch.rs

```diff
@@ -1,191 +1,106 @@
+        return Err(anyhow::anyhow!("File not found: {}", file));
     }
+    Ok(())
 }
 
-#[derive(Debug)]
-struct Payment {
-    username: String,
-    usd: Decimal,
-    sats: Option<Decimal>,
-    wallet_id: Option<String>,
-    memo: Option<String>,
-}
-
-pub struct Batch {
-    payments: Vec<Payment>,
-    client: GaloyClient,
-    /// price in btc/usd
-    price: Decimal,
-}
-
-impl Batch {
-    pub fn new(client: GaloyClient, price: Decimal) -> Self {
-        let payments: Vec<Payment> = vec![];
-        Self {
-            payments,
-            client,
-            price,
-        }
+// Utility function to read and validate the CSV file
+pub fn validate_csv(
+    galoy_cli: &GaloyClient,
+    file: &str,
+) -> anyhow::Result<(Vec<csv::StringRecord>, Wallet)> {
+    let mut reader = csv::ReaderBuilder::new().delimiter(b',').from_path(file)?;
+    let headers = reader.headers()?.clone();
+
+    if &headers[0] != "username"
+        || (&headers[1] != "cents" && &headers[1] != "sats")
+        || (headers.len() == 3 && &headers[2] != "memo")
+    {
+        return Err(anyhow::anyhow!(
+            "CSV format not correct, requires: username, (cents or sats), memo(optional)"
+        ));
     }
 
-    pub fn add(&mut self, input: PaymentInput) {
-        self.payments.push(input.into());
-    }
-
-    pub fn add_csv(&mut self, filename: String) -> anyhow::Result<()> {
-        let file = File::open(filename)?;
-        let mut rdr = csv::Reader::from_reader(file);
-        for result in rdr.deserialize() {
-            let record: PaymentInput = result?;
-            self.add(record);
-        }
-
-        Ok(())
-    }
-
-    pub fn len(&self) -> usize {
-        self.payments.len()
-    }
-
-    pub fn is_empty(&self) -> bool {
-        self.payments.is_empty()
-    }
+    let wallet_type = if headers.get(1) == Some("cents") {
+        Wallet::Usd
+    } else {
+        Wallet::Btc
+    };
 
-    pub fn populate_wallet_id(&mut self) -> anyhow::Result<()> {
-        for payment in self.payments.iter_mut() {
-            let username = payment.username.clone();
-            let query = &self.client.default_wallet(username);
-            match query {
-                Ok(value) => payment.wallet_id = Some(value.clone()),
-                Err(error) => bail!("error query {:?}", error),
-            }
-        }
+    let records: Vec<csv::StringRecord> = reader.records().collect::<Result<_, _>>()?;
 
-        Ok(())
-    }
+    // Validate each record
+    for record in &records {
+        let username = record
+            .get(0)
+            .ok_or(anyhow::anyhow!("Username is missing"))?;
 
-    pub fn populate_sats(&mut self) -> anyhow::Result<()> {
-        for payment in self.payments.iter_mut() {
-            let payment_btc: Decimal = payment.usd / self.price;
-            payment.sats = Some(payment_btc * dec!(100_000_000));
-        }
+        let amount = record.get(1).ok_or(anyhow::anyhow!("Amount is missing"))?;
```

Comment:
I find the syntax:
```
 record.get(1).context("Amount is missing")?;
 ```
 more concise.
 Need to `use anyhow::Context` for it to work.